src/typing_extensions.py

```diff
@@ -1282,7 +1282,7 @@ def _clean_optional(obj, hints, globalns=None, localns=None):
             ):
                 continue
             original_value = original_hints[name]
-            if original_value is None:  # should not happen
+            if original_value is None:  # should be NoneType already; check just in case
```

Comment:
It shouldn't be NoneType already, since we get this directly from `__annotations__`.

Reply:
Excuse me I have argued wrongly. You are right that `None` would be from `__annotations__`. In such a case `_could_be_inserted_optional(NoneType)` would have already skipped this iteration (`get_origin(NoneType) is not Union`). That's why `original_value` should be never None here.